lib/api/BackbeatAPI.js

```diff
@@ -1383,14 +1253,47 @@ class BackbeatAPI {
             }),
         ], err => {
             if (err) {
-                this._logger.error('error setting up internal clients');
+                this._logger.error('error setting up internal clients', {
+                    method: 'BackbeatAPI.setupInternals',
+                    error: err.message,
+                });
                 return cb(err);
             }
             this._healthcheck = new Healthcheck(this._repConfig, this._zkClient,
                 this._crrProducer, this._crrStatusProducer,
                 this._metricProducer);
-            this._logger.info('BackbeatAPI setup ready');
-            return cb();
+            this._locationStatusManager = new LocationStatusManager(
+                mongoLocationStatusServices,
+                this._mongoClient,
+                this._zkClient,
+                this._redisPublisher,
+                {
+                    crr: this._crrTopic,
+                    ingestion: this._ingestionTopic,
+                },
+                {
+                    crr: {
+                        namespace: zookeeperReplicationNamespace,
+                        statePath: zkReplicationStatePath,
+                    },
+                    ingestion: {
+                        namespace: zookeeperIngestionNamespace,
+                        statePath: zkIngestionStatePath,
+                    },
+                },
```

Comment:
nit: maybe we should merge the "topics" and "zookeeper" params in a single map:

```suggestion
                {
                    crr: {
                        topic: this._crrTopic,
                        namespace: zookeeperReplicationNamespace,
                        statePath: zkReplicationStatePath,
                    },
                    ingestion: {
                        topic: this._ingestionTopic,
                        namespace: zookeeperIngestionNamespace,
                        statePath: zkIngestionStatePath,
                    },
                },
```

Reply:
and also merge the information from `mongoLocationStatusServices`, somehow: e.g. should have an entry to the service even if it is mongo, though in that case it would NOT have the redis topic or zookeeper state.